tests/bsim/bluetooth/host/l2cap/einprogress/dut/src/dut.c

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/l2cap.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/net/buf.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/sys/util_macro.h>
+
+#include "testlib/conn.h"
+#include "testlib/scan.h"
+
+#include "babblekit/flags.h"
+#include "babblekit/testcase.h"
+
+/* local includes */
+#include "data.h"
+
+LOG_MODULE_REGISTER(dut, CONFIG_APP_LOG_LEVEL);
+
+struct tester_t {
```

Comment:
What's the `_t` in `tester_t`?

Reply:
This is copy-pasted code from a different test. There it was just `tester`. But the CI now dislikes overlap in the struct type namespace and variable namespace. I just added `_t` for "type". Any other ideas?

Reply:
> I just added _t for "type"

Our coding guidelines prohibit that :) 

Could just be something like `tester_data` or `tester_inst` or something

Reply:
> > I just added _t for "type"
> 
> Our coding guidelines prohibit that :)

Just to follow up on this. We don have a convention of `_t` but that's for typedefs, like `bt_addr_le_t`. With a struct name you either way have to do `struct foo` whenever you reference the type, so it's redundant to put any additional stuff in the name itself.

Reply:
Where is that rule documented?
https://kernel.org/doc/html/latest/process/coding-style.html#typedefs only talks about typedefs vs `struct foo`, not the `struct foo_t` being illegal.

Reply:
> Where is that rule documented?

Not sure if that was a question to me or @Thalley, but I don't think I said "rule" anywhere. I was only talking about the existing convention (as well as explaining the rationale behind it). That said, do we really need to have an explicitly documented "Be consistent with existing code" rule? I'd have hoped that it goes without saying.

Reply:
> Where is that rule documented? https://kernel.org/doc/html/latest/process/coding-style.html#typedefs only talks about typedefs vs `struct foo`, not the `struct foo_t` being illegal.

Indeed. It only applies to `typedefs` as has been pointed out. I made a mistake in assuming that the same idea would also apply for structs. 

> That said, do we really need to have an explicitly documented "Be consistent with existing code" rule? I'd have hoped that it goes without saying.

We actually used to have that somewhere IIRC back in 2018-2019-ish, but I'm failing to find it now.

Reply:
> Be consistent with existing code

Since this existing code style differs between subsystems and sometimes even between individual units, I would like to have a document stating what the policy is.

I'm being obtuse on purpose, because the process people are burying their head in the sand with #75772. If we had guidelines, this whole thread wouldn't exist.